### PR TITLE
[Localization] Added german localization for challenges

### PR DESCRIPTION
**Pull request by CodeTappert**
## What are the changes?

German Localization for the challenges

## Why am I doing these changes?

Because they werent localized

## What did change?

I translated all the keys into german

### Screenshots/Videos

image
image

## How to test the changes?

Set your game to german and startup a new challenge

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`) 
    - They are also failing on main. And not with anything i changed
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/de/challenges.ts [hunk 1]
```diff
@@ -1,67 +1,67 @@
...
-  "singleGeneration.desc.7": "You can only use pokemon from generation seven.",
-  "singleGeneration.value.8": "Gen 8",
-  "singleGeneration.desc.8": "You can only use pokemon from generation eight.",
-  "singleGeneration.value.9": "Gen 9",
-  "singleGeneration.desc.9": "You can only use pokemon from generation nine.",
-  "singleType.name": "Mono Type",
-  "singleType.value.0": "Off",
-  "singleType.desc.0": "You can only use pokemon of the chosen type.",
+  "title": "Herausforderungsmodifikatoren",
+  "confirm_start": "Mit diesen Einschränkungen fortfahren?",
```
Comment (flx-sta) [inline src/locales/de/challenges.ts:5]: Wieso “Einschränkungen” und nicht “Modifikationen”? Steht ja auch so im Titel

**Review by flx-sta — CHANGES_REQUESTED** [review]

File: src/locales/de/challenges.ts [hunk 2]
```diff
@@ -1,67 +1,67 @@
...
-  "singleGeneration.value.7": "Gen 7",
-  "singleGeneration.desc.7": "You can only use pokemon from generation seven.",
-  "singleGeneration.value.8": "Gen 8",
-  "singleGeneration.desc.8": "You can only use pokemon from generation eight.",
-  "singleGeneration.value.9": "Gen 9",
-  "singleGeneration.desc.9": "You can only use pokemon from generation nine.",
-  "singleType.name": "Mono Type",
-  "singleType.value.0": "Off",
-  "singleType.desc.0": "You can only use pokemon of the chosen type.",
+  "title": "Herausforderungsmodifikatoren",
```
[the remark this answers is not in the document]
Comment (CodeTappert) [reply src/locales/de/challenges.ts:4]: Xavion said it was not needed

File: src/locales/de/challenges.ts (hunk 1 above)
Comment (CodeTappert) [reply src/locales/de/challenges.ts:5]: Make a suggestion for that change please

Comment (flx-sta) [reply src/locales/de/challenges.ts:5]:
```suggestion
  "confirm_start": "Mit diesen Modifikatoren fortfahren?",
```

**Comment by flx-sta** [comment]
So this is not part of THIS PR but I have a question to the person originally developing it:

why is there a translation for every single type necessary ? We could have just used a placeholder like:
```ts
You can only pick Pokemon of type {{type}}
```

a single translation would have covered them all then

File: src/locales/de/challenges.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (flx-sta) [reply src/locales/de/challenges.ts:4]: Might wanna delete the English one too then ?

**Review by FredeX — APPROVED** [review]
Lgtm

**Review by flx-sta — APPROVED** [review]

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**